easypy/caching.py

```diff
@@ -179,12 +180,14 @@ def inner(*args, **kwargs):
                 result, ts = cache.get(key, NOT_CACHED)
 
                 if expiration > 0:
-                    if result is not NOT_FOUND and time.time() - ts > expiration:
+                    if result is not NOT_FOUND and get_ts_func() - ts > expiration:
                         result = NOT_FOUND
                         del cache[key]
 
                 if result is NOT_FOUND:
-                    ts = time.time()
+                    if log_recalculation:
+                        _logger.debug('time cache expired, calculating new value for %s', name)
+                    ts = get_ts_func()
```

Comment:
What @idanarye probably means (and I agree if this what he means) - It's more correct to move `ts = get_ts_func()` under `result = func(*args, **kwargs)`

So timestamp will represent when result was generated, not when it began to generate.  

And for a same reason it's incorrect to put `now = get_ts_func()` in the beginning of function. Validation and generation timestamps might differ